workbench-core/datasets/src/dataSetService.ts

```diff
@@ -245,16 +245,32 @@ export class DataSetService {
    *
    * @returns an array of DataSet objects.
    */
-  public async listDataSets(authenticatedUser: { id: string; roles: string[] }): Promise<DataSet[]> {
+  public async listDataSets(authenticatedUser: AuthenticatedUser): Promise<DataSet[]> {
     const metadata: Metadata = {
       actor: authenticatedUser,
       action: this.listDataSets.name,
       source: {
         serviceName: DataSetService.name
       }
     };
+
     try {
-      const response = await this._dbProvider.listDataSets();
+      const response: DataSet[] = [];
+      const allDatasets = await this._dbProvider.listDataSets();
+
+      const allDatasetsPermissions = await Promise.all(
+        allDatasets.map((dataset) =>
+          this._getAuthenticatedUserDatasetPermissions(authenticatedUser, dataset.id)
```

Comment:
why not just call isAuthorizedOnSubject() for these?

Reply:
That function isnt available through the `DataSetsAuthorizationPlugin` so I cant use it.